tests/python/contrib/test_arm_compute_lib/test_add.py

```diff
@@ -0,0 +1,142 @@
+    a = relay.var(next(var_names), shape=shape, dtype=dtype)
+    b = relay.var(next(var_names), shape=shape, dtype=dtype)
+    return op(a, b, **op_params)
+
+
+def _get_expected_codegen(shape, dtype):
+    node = {
+        "op": "kernel",
+        "name": "add",
+        "inputs": [[0, 0, 0], [1, 0, 0]],
+        "attrs": {
+            "num_inputs": "2",
+            "num_outputs": "1",
+            "shape": [[list(shape)]],
+            "dtype": [[dtype]],
+        },
+    }
+
+    input_a = {"op": "input", "name": "", "attrs": {"shape": [[list(shape)]], "dtype": [[dtype]]}}
+    input_b = {"op": "input", "name": "", "attrs": {"shape": [[list(shape)]], "dtype": [[dtype]]}}
+    return [input_a, input_b, node]
+
+
+def test_add():
+    Device.load("test_config.json")
+
+    if skip_runtime_test():
+        return
+
+    device = Device()
+    np.random.seed(0)
+
+    for dtype, low, high, atol, rtol, op, op_params in [
+        ("float32", -127, 128, 1e-7, 1e-7, relay.add, {}),
+        # different qnn params
+        (
+            "uint8",
+            0,
+            255,
+            0.0,
+            1.0,
+            relay.qnn.op.add,
+            {
+                "lhs_scale": relay.const(0.0156863, "float32"),
+                "lhs_zero_point": relay.const(127, "int32"),
+                "rhs_scale": relay.const(0.0117647, "float32"),
+                "rhs_zero_point": relay.const(85, "int32"),
+                "output_scale": relay.const(0.0235294, "float32"),
+                "output_zero_point": relay.const(128, "int32"),
+            },
+        ),
+        # same qnn params
+        (
+            "uint8",
+            0,
+            255,
+            0.0,
+            1.0,
+            relay.qnn.op.add,
+            {
+                "lhs_scale": relay.const(0.0126863, "float32"),
+                "lhs_zero_point": relay.const(127, "int32"),
+                "rhs_scale": relay.const(0.0126863, "float32"),
+                "rhs_zero_point": relay.const(127, "int32"),
+                "output_scale": relay.const(0.0126863, "float32"),
+                "output_zero_point": relay.const(127, "int32"),
+            },
+        ),
+    ]:
+        shape = (2, 2)
+        for inputs in [
+            {
+                "a": tvm.nd.array(np.random.uniform(low, high, shape).astype(dtype)),
+                "b": tvm.nd.array(np.random.uniform(low, high, shape).astype(dtype)),
+            }
+        ]:
+            outputs = []
+            func = _get_model(shape, dtype, iter(inputs), op, op_params)
+            for acl in [True, False]:
+                outputs.append(build_and_run(func, inputs, 1, None, device, enable_acl=acl)[0])
+
+            config = {
+                "shape": shape,
+                "dtype": dtype,
+                "inputs": inputs,
+                "operation": op,
+                "op_params": op_params,
+            }
+            verify(outputs, atol=atol, rtol=rtol, config=config)
```

Comment:
When we test qnn.add we should also use the parameter `verify_saturation=True`

Reply:
Could you elaborate the assertion of "0.25 * outs[0].asnumpy().size" for small sized (e.g. (2,2) or similar ) test cases, please?

Reply:
The intention was to ensure that most of the values output are not saturated i.e. most values are not 255 or 0. I can see why this would cause an issue with test cases of 2x2 as a single value that is 255 or 0 would trigger the asserts. I think the best approach here would be to increase the size of the test case?

Reply:
add_QASYMM8_QASYMM8_QASYMM8 currently always saturate result. So verify_saturation=True cannot be used